[E2E][GB][getTestAccountByFeature] Add `envToFeatureKey` helper

fullofcaffeine opened the pull request:
Follow up to: https://github.com/Automattic/wp-calypso/pull/61783.
Project: p9oQ9f-13V-p2.

#### Changes proposed in this Pull Request

* Add `envToFeatureKey` (ad-hoc) helper to turn a `SupportedEnvVariables`ish object into a `FeatureKey`. 

##### Additional changes

* Move the `defaultCriteria` definition to its own module;
* Fixed a [test](https://github.com/Automattic/wp-calypso/pull/62192/files#diff-299c981fa7af6648cb08a17e29e87c51177f8ab7944d4aa8f05abe801b66b556R100).


#### Other observations

I initially pondered if we could just change the `getTestAccountByFeature` signature to accept a single `SupportedEnvVariable`-like object (like @dpasque suggested [here](https://github.com/Automattic/wp-calypso/pull/61783#discussion_r827378857)), or create a public wrapper that would accept an env var object instead of a `FeatureKey` object. If _all_ input data indeed came from env variables, it'd make sense to do it, while making the latter part of the encapsulate private implementation detail. 

However, there will be cases when we'd like to pass a test-specific feature criterion that's not dependent on an external env var. A clear example of that is the optional `variant` criterion that can be passed as part of a `FeatureKey`. It's very test-specific and can't be set through an env var (an env var is by definition build-specific) and defines if the test is for the `siteEditor` instead of the regular editor.[ Example here](https://github.com/Automattic/wp-calypso/pull/62192/files#diff-f0bf6913db82b58ce87f903fa5116e2b7c71306b8bc321cfe93dc1092c07dcbaR29). Hence, it looks like it's best to keep `envToFeatureKey' as a public helper function that can be used inline whenever needed and its return value can be extended inline, if needed, too. 

#### Testing instructions

* Tests should pass.

fullofcaffeine committed:
Add `envToFeatureKey` helper function to be used with `getAccountByFeature`

This function effectively converts an `SupportedEnvVariable`ish object into a `FeatureKey` that
can be passed to the `getAccountByFeature` function, allowing to DRY the its call even further.

fullofcaffeine committed:
Migrate instances of `getTestAccountByFeature` across tests to use `envToFeatureKey`

fullofcaffeine committed:
Fix eslint issues

dpasque commented on packages/calypso-e2e/src/lib/utils/get-test-account-by-feature.ts:
```diff
@@ -132,3 +93,30 @@ export function getTestAccountByFeature(
+ * often, changes to the former might require the logic here to be updated, so
+ * beware :)
+ *
+ * @param {SupportedEnvVariables} envVariables
```
Tiny nit - I think you have the wrong type here in the JSDOC comment!

fullofcaffeine replied:
Ah, true! Thanks for catching that!

dpasque reviewed (approved):
Beautiful!! 😻  One tiny nit, otherwise LGTM! 🚀

fullofcaffeine committed:
Connect the `target` `Editor` param to the `TEST_ON_ATOMIC` env var via the `FeatureKey` object's `siteType` attribute (#62196)

To enable these tests to use the slightly different `Editor` implementation for AT site (i.e no Gutenframe) when the build is setup to do so (`env.TEST_ON_ATOMIC` is set).

fullofcaffeine committed:
Add stable and edge AT E2E WP accounts for testing the Site Editor (#62195)

fullofcaffeine committed:
Fix type in jsdoc
